examples/direct_fidelity_estimation.py

```diff
@@ -29,27 +30,44 @@
 import cirq
 
 
-def build_circuit():
-    # Builds an arbitrary circuit to test. The circuit is non Clifford to show
-    # the use of simulators.
-    qubits = cirq.LineQubit.range(3)
-    circuit = cirq.Circuit(
-        cirq.Z(qubits[0])**0.25,  # T-Gate, non Clifford.
-        cirq.X(qubits[1])**0.123,
-        cirq.X(qubits[2])**0.456)
+def build_circuit(
+) -> Tuple[cirq.Circuit, List[cirq.Qid], Optional[cirq.CliffordState]]:
+    # Builds an arbitrary circuit to test. If the circuit is Clifford, then it
+    # also returns a state from which we can get the stabilitzers. If the
+    # circuit is non Clifford, the state is None.
+    qubits: List[cirq.Qid] = [qubit for qubit in cirq.LineQubit.range(3)]
+    circuit: cirq.Circuit = cirq.Circuit()
+    clifford_state = cirq.CliffordState(
+        qubit_map={qubits[i]: i for i in range(len(qubits))})
+    is_clifford = True
+    gates = [
+        cirq.Z(qubits[0]),
+        cirq.X(qubits[1]),
+        cirq.X(qubits[2]),
+    ]
+    for gate in gates:
+        circuit.append(gate)
+        if is_clifford:
+            try:
+                clifford_state.apply_unitary(gate)
+            except ValueError:
+                print('Circuit is non Clifford.')
+                is_clifford = False
     print('Circuit used:')
     print(circuit)
-    return circuit, qubits
+    if is_clifford:
+        return circuit, qubits, clifford_state
+    else:
+        return circuit, qubits, None
```

Comment:
likewise, different return types is a code smell. I would suggest always returning a clifford state. Maybe you can have another function that returns a non-clifford circuit and a flag somewhere else to switch between them. This would probably require factoring out the "check if circuit is clifford" logic elsewhere

Reply:
No longer present. Done.